src/lib/EventPublisher.js

```diff
@@ -140,25 +151,30 @@ class EventPublisher extends EventEmitter {
           timeoutHandler();
         }
       }, subscriptionTimeout);
-
-      this.once(eventNamePrefix, (): void => clearTimeout(timeout));
+      this.once(eventNamePrefix, () => {
+        clearTimeout(timeout);
+      });
     }
 
     if (once) {
       this.once(eventNamePrefix, listener);
+      this.once(eventNamePrefix, () => {
+        this._subscriptionsByID.delete(subscriptionID);
```

Comment:
Help me to understand the problem:

in case of once, there are 2 functions subscribed which will be remove automatically (once). first is the listener, second is removing the subscribtion...

What's wrong with this

```
this.once(eventNamePrefix, listener);
this.once(eventNamePrefix, () => {
        this._subscriptionsByID.delete(subscriptionID);
});
```